src/ape/managers/chain.py

```diff
@@ -516,12 +516,36 @@ def cache_deployment(self, contract_instance: ContractInstance):
               to cache.
         """
 
+        proxy_info = self.provider.network.ecosystem.get_proxy_info(contract_instance.address)
+        if proxy_info:
+            self.cache_proxy_info(contract_instance.address, proxy_info)
+            return
+
         address = contract_instance.address
         contract_type = contract_instance.contract_type
         txn_hash = contract_instance.txn_hash
         self._cache_contract_type(address, contract_type)
         self._cache_deployment(address, contract_type, txn_hash)
 
+    def cache_proxy_info(self, address: AddressType, proxy_info: ProxyInfoAPI):
+        """
+        Cache proxy info for a particular address, useful for plugins adding already
+        deployed proxies. When you deploy a proxy locally, it will also call this method.
+
+        Args:
+            address (AddressType): The address of the proxy contract.
+            proxy_info (:class:`~ape.api.networks.ProxyInfo`): The proxy info class
+              to cache.
+        """
+
+        if self.get_proxy_info(address) and self._is_live_network:
+            return
```

Comment:
why not cache if we have the proxy info and this is a live network?

Reply:
Because we should cache the target contract type and not the proxys, but I just realized I forgot that part!

- [ ] cache target ct